griptape/tools/openweather_client/tool.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import annotations
+from griptape.artifacts import ListArtifact, TextArtifact, ErrorArtifact
+from griptape.tools import BaseTool
+from griptape.utils.decorators import activity
+from schema import Schema, Literal
+from attr import define, field
+from typing import Optional
+import requests
+import logging
+
+@define
+class OpenWeatherClient(BaseTool):
+    BASE_URL = "https://api.openweathermap.org/data/3.0/onecall"
+    GEOCODING_URL = "https://api.openweathermap.org/geo/1.0/direct"
+    api_key: str = field(kw_only=True)
+    units: Optional[str] = field(default="imperial", kw_only=True)
+
+    @activity(
+        config={
+            "description": "Fetches the latitude and longitude for a given location using the OpenWeather Geocoding API.",
```

Comment:
1. Make sure it aligns with other tool activity descriptions in voice. I.e., "Can be used..."
2. No need to provide details of what API is used. Just describe the function of the activity.